olp-cpp-sdk-core/tests/logging/ConfigurationTest.cpp

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (C) 2020 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gtest/gtest.h>
+#include <olp/core/logging/Configuration.h>
+#include <olp/core/logging/Log.h>
+#include <memory>
+#include "MockAppender.h"
+
+namespace {
+
+using namespace olp::logging;
+using namespace testing;
+
+TEST(ConfigurationTest, InvalidConfiguration) {
+  Configuration configuration;
+  EXPECT_FALSE(configuration.isValid());
+  EXPECT_FALSE(Log::configure(configuration));
+}
+
+TEST(ConfigurationTest, DefaultConfiguration) {
+  Configuration configuration = Configuration::createDefault();
+  EXPECT_TRUE(configuration.isValid());
+  EXPECT_TRUE(Log::configure(configuration));
+}
+
+TEST(ConfigurationTest, MockAppender) {
+  Configuration configuration;
+  configuration.addAppender(std::make_shared<MockAppender>());
```

Comment:
Would it not bet better to mock appender with gmock EXPECT_CALLs and realy check here that the Append() method is called when log message is written?

Reply:
The approach is rather simple - to access and to check appended messages directly after invoking logging. That should be simpler/cleaner than the gmock approach.

Reply:
Also +1 to this, all the tests, that are imported, use the MockAppender

Reply:
_Resolved offline._
Agreed that we stick with the MockAppender class and in the future, if the benefits will overweight the effort, we'll switch to using `gmock` in the `logging` tests